ostd/src/arch/x86/iommu/dma_remapping/second_stage.rs

```diff
@@ -82,7 +82,15 @@ impl PageTableEntryTrait for PageTableEntry {
     }
 
     fn new_pt(paddr: Paddr) -> Self {
-        Self(paddr as u64 & Self::PHYS_MASK)
+        // FIXME: The IOMMU page table will check the permission of the PTE that maps to a child
+        // page table. Here is a workaround to make sure devices can access memory. The devices
+        // are still limited by the permissions of the underlying mapping, so this workaround
+        // does not result in undefined behavior.
```

Comment:
In short: I think it won't be "FIXED" so "FIXME" is not needed here.

Not only the IOMMU page table checks the permissions to an intermediate page table, the normal page tables do too. I do not think it a workaround because it leads to no problems. Indeed on x86 platforms we can mark the permissions of an intermediate page table entry as the union of the child's permissions. But this seems to complicate the book-keeping and doesn't really remediating the TLB miss penalty by much.

Actually I am thinking that the `PageTableEntryTrait` should be an unsafe trait, since it's implementation affects memory safety.